module/importer-magento-1/src/Infrastructure/Reader/Magento1CsvReader.php

```diff
@@ -26,106 +25,171 @@ class Magento1CsvReader
     private ConverterMapperProvider $mapper;
 
     /**
-     * @var ReaderProcessorProvider
+     * @var string
      */
-    private ReaderProcessorProvider $provider;
+    private string $directory;
 
     /**
-     * @var string
+     * @var mixed
      */
-    private string $directory;
+    private $file;
+
+    /**
+     * @var array
+     */
+    private array $headers;
+
 
     /**
      * @param ConverterMapperProvider $mapper
-     * @param ReaderProcessorProvider $provider
      * @param string                  $directory
      */
-    public function __construct(
-        ConverterMapperProvider $mapper,
-        ReaderProcessorProvider $provider,
-        string $directory
-    ) {
+    public function __construct(ConverterMapperProvider $mapper, string $directory)
+    {
         $this->mapper = $mapper;
-        $this->provider = $provider;
         $this->directory = $directory;
     }
 
     /**
-     * @param Magento1CsvSource $source
-     * @param Import            $import
-     * @param Transformer       $transformer
-     *
-     * @return array
+     * @param Import $import
      */
-    public function read(Magento1CsvSource $source, Import $import, Transformer $transformer): array
+    public function open(Import $import): void
     {
-        $errors = [];
-        $filename = \sprintf('%s%s', $this->directory, $import->getFile());
-        $extension = pathinfo($filename, PATHINFO_EXTENSION);
-        $fileReader = $this->provider->provide($extension);
+        $file = \sprintf('%s%s', $this->directory, $import->getFile());
 
-        $products = [];
-        $sku = null;
-        $type = null;
+        try {
+            $this->file = \fopen($file, 'rb');
```

Comment:
I strongly suggest not to use internal runtime state(file, headers), it's way better to just pass variables internally(alternatively an object can be created with factory). Easily can cause inconsistencies in the future